diff --git a/runner/core-module.vala b/runner/core-module.vala
index 0bf3446..874b57f 100644
--- a/runner/core-module.vala
+++ b/runner/core-module.vala
@@ -18,14 +18,14 @@ private class Highscore.CoreModule : TypeModule {
 
         try {
             var keyfile_path = Path.build_filename (
-                Config.CORES_DIR,
+                Highscore.CORES_DIR,
                 @"$core_filename.highscore"
             );
 
             var keyfile = new KeyFile ();
             keyfile.load_from_file (keyfile_path, KeyFileFlags.NONE);
             var module_name = keyfile.get_string ("Highscore", "Module");
-            module_path = Path.build_filename (Config.CORES_DIR, module_name);
+            module_path = Path.build_filename (Highscore.CORES_DIR, module_name);
             name = keyfile.get_string ("Highscore", "Name");
 
             if (keyfile.has_group ("Hacks")) {
diff --git a/runner/main.vala b/runner/main.vala
index 32c6ccb..733ef04 100644
--- a/runner/main.vala
+++ b/runner/main.vala
@@ -2,7 +2,12 @@
 
 using Highscore;
 
+namespace Highscore {
+    public string CORES_DIR;
+}
 int main (string[] args) {
+    Highscore.CORES_DIR = Path.get_dirname(args[0]) + "/../lib/highscore/cores";
+    debug ("Highscore.CORES_DIR: %s", Highscore.CORES_DIR);
     Intl.bindtextdomain (Config.GETTEXT_PACKAGE, Config.GNOMELOCALEDIR);
     Intl.bind_textdomain_codeset (Config.GETTEXT_PACKAGE, "UTF-8");
     Intl.textdomain (Config.GETTEXT_PACKAGE);
diff --git a/src/core-register.vala b/src/core-register.vala
index bc6da61..368ee2e 100644
--- a/src/core-register.vala
+++ b/src/core-register.vala
@@ -21,7 +21,7 @@ public class Highscore.Core : Object {
 
         try {
             var keyfile_path = Path.build_filename (
-                Config.CORES_DIR,
+                Highscore.CORES_DIR,
                 @"$core_filename.highscore"
             );
 
@@ -60,7 +60,7 @@ public class Highscore.CoreRegister : Object {
         cores = new HashTable<string, Core> (str_hash, str_equal);
 
         try {
-            var directory = File.new_for_path (Config.CORES_DIR);
+            var directory = File.new_for_path (Highscore.CORES_DIR);
             var enumerator = directory.enumerate_children (FileAttribute.STANDARD_NAME, 0);
 
             FileInfo info;
diff --git a/src/main.vala b/src/main.vala
index 24fe0b1..31a662b 100644
--- a/src/main.vala
+++ b/src/main.vala
@@ -1,6 +1,14 @@
 // This file is part of Highscore. License: GPL-3.0-or-later
 
+namespace Highscore {
+    public string CORES_DIR;
+    public string RUNNER_PATH;
+}
 int main (string[] args) {
+    Highscore.CORES_DIR = Path.get_dirname(args[0]) + "/../lib/highscore/cores";
+    Highscore.RUNNER_PATH = Path.get_dirname(args[0]) + "/../libexec/highscore-runner";
+    debug ("Highscore.CORES_DIR: %s", Highscore.CORES_DIR);
+    debug ("Highscore.RUNNER_PATH: %s", Highscore.RUNNER_PATH);
     Intl.bindtextdomain (Config.GETTEXT_PACKAGE, Config.GNOMELOCALEDIR);
     Intl.bind_textdomain_codeset (Config.GETTEXT_PACKAGE, "UTF-8");
     Intl.textdomain (Config.GETTEXT_PACKAGE);
diff --git a/src/runner/runner-process.vala b/src/runner/runner-process.vala
index c641344..12e750f 100644
--- a/src/runner/runner-process.vala
+++ b/src/runner/runner-process.vala
@@ -79,7 +79,7 @@ public class Highscore.RunnerProcess : Object {
         if (runner_path_override == null)
             runner_path_override = Environment.get_variable ("HIGHSCORE_RUNNER");
 
-        var path = runner_path_override ?? Config.RUNNER_PATH;
+        var path = runner_path_override ?? Highscore.RUNNER_PATH;
 
         string[] args = {};
 
